IGI.Lab2/src/CarPark.DAL/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Collections.Generic;
+
+namespace CarPark.DAL.Interfaces
+{
+    public interface IRepository <TEntity> where TEntity : class, IEntity
+    {
+        TEntity Get(int id);
```

Comment:
Looks great, but all these methods should be asynchronous.

Reply:
fixed